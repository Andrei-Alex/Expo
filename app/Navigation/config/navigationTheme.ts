import { DefaultTheme } from '@react-navigation/native';
import { colors } from '../../config';

export const appTheme = {
  ...DefaultTheme,
  colors: {
    ...DefaultTheme.colors,
    primary: colors.primary,
    background: colors.white
  }
};
