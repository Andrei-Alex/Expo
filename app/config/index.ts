export { default as colors } from './theme/colors';
