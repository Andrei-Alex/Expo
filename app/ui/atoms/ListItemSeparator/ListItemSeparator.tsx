import React from 'react';
import { View } from 'react-native';

import { styles, Props } from '.';

const ListItemSeparator: React.FC<Props> = ({ style }) => {
  return <View style={[styles.separator, style]}></View>;
};

export default ListItemSeparator;
