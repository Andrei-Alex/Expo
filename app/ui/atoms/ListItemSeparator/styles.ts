import { StyleSheet } from 'react-native';
import { colors } from '../../../config';

export const styles = StyleSheet.create({
  separator: {
    width: '100%',
    height: 1,
    backgroundColor: colors.light
  }
});
