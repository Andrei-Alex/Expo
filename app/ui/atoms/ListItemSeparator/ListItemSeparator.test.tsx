import * as React from 'react';
import { View } from 'react-native';
import * as renderer from 'react-test-renderer';

import { colors } from '../../../config';
import ListItemSeparator from './ListItemSeparator';

describe('AppText', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ListItemSeparator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('applies passed ViewStyle', () => {
    const styles = { backgroundColor: colors.danger };

    const instance = renderer.create(<ListItemSeparator style={styles} />).root;
    const viewComponent = instance.findByType(View);

    const receivedStyle = viewComponent.props.style;
    const styleArray = Array.isArray(receivedStyle)
      ? receivedStyle
      : [receivedStyle];

    const hasTestStyle = styleArray.some((style) =>
      Object.entries(style).every(([key, value]) => style[key] === value)
    );
    expect(hasTestStyle).toBeTruthy();
  });
});
