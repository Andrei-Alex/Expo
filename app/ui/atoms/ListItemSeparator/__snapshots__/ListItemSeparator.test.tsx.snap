// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppText renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#f8f4f4",
        "height": 1,
        "width": "100%",
      },
      undefined,
    ]
  }
/>
`;
